synapse/storage/util/id_generators.py

```diff
@@ -448,6 +448,9 @@ def get_next_txn(self, txn: LoggingTransaction):
         # position points to a persisted row with the correct instance name.
         if self._writers:
             txn.call_after(
+                # Argument 1 to "call_after" of "LoggingTransaction" has incompatible
+                # type "Callable[[str, Callable[..., Awaitable[R]], VarArg(Any), DefaultNamedArg(bool, 'bg_start_span'), KwArg(Any)], Deferred[R]]";
+                # expected "Callable[..., None]"  [arg-type]
```

Comment:
`run_as_background_process` returns a `Deferred[None]`, but `call_after` expects a function returning `None`

Shall we update the types of `call_after` to allow functions returning anything?
Or wrap this in a function returning `None`?